src/environment.rs

```diff
@@ -450,6 +451,12 @@ impl Environment {
         self.threads[self.current_thread].blocked_by = ThreadBlock::Mutex(mutex_id);
     }
 
+    /// Blocks the current thread until the thread given finishes, writing it's return value to ptr
+    /// (if non-null).
+    /// Note that there are no protections against joining with a deattached thread, joining a thread with itself,
```

Comment:
Nit: deattached -> detached.

Reply:
Oh also, FWIW I prefer it if comments don't go past the 80-column mark. I should really set up something to detect and enforce this kind of thing, it's a waste of time to enforce this manually.